flink-tests/src/test/java/org/apache/flink/test/runtime/ShuffleCompressionITCase.java

```diff
@@ -0,0 +1,173 @@
+	}
+
+	@Test
+	public void testDataCompressionForBlockingShuffle() throws Exception {
+		executeTest(createJobGraph(ScheduleMode.LAZY_FROM_SOURCES, ResultPartitionType.BLOCKING, ExecutionMode.BATCH));
+	}
+
+	private void executeTest(JobGraph jobGraph) throws Exception {
+		Configuration configuration = new Configuration();
+		configuration.setString(TaskManagerOptions.TOTAL_FLINK_MEMORY, "1g");
+		configuration.setBoolean(NettyShuffleEnvironmentOptions.BLOCKING_SHUFFLE_COMPRESSION_ENABLED, true);
+		configuration.setBoolean(NettyShuffleEnvironmentOptions.PIPELINED_SHUFFLE_COMPRESSION_ENABLED, true);
+
+		final MiniClusterConfiguration miniClusterConfiguration = new MiniClusterConfiguration.Builder()
+			.setConfiguration(configuration)
+			.setNumTaskManagers(PARALLELISM)
+			.setNumSlotsPerTaskManager(1)
+			.build();
+
+		try (MiniCluster miniCluster = new MiniCluster(miniClusterConfiguration)) {
+			miniCluster.start();
+
+			MiniClusterClient miniClusterClient = new MiniClusterClient(configuration, miniCluster);
+			// wait for the submission to succeed
+			JobID jobID = miniClusterClient.submitJob(jobGraph).get();
+
+			CompletableFuture<JobResult> resultFuture = miniClusterClient.requestJobResult(jobID);
+			assertFalse(resultFuture.get().getSerializedThrowable().isPresent());
+		}
+	}
+
+	private static JobGraph createJobGraph(
+			ScheduleMode scheduleMode,
+			ResultPartitionType resultPartitionType,
+			ExecutionMode executionMode) throws IOException {
+		SlotSharingGroup slotSharingGroup = new SlotSharingGroup();
+
+		JobVertex source = new JobVertex("source");
+		source.setInvokableClass(LongValueSource.class);
+		source.setParallelism(PARALLELISM);
+		source.setSlotSharingGroup(slotSharingGroup);
+
+		JobVertex sink = new JobVertex("sink");
+		sink.setInvokableClass(ResultVerifyingSink.class);
+		sink.setParallelism(PARALLELISM);
+		sink.setSlotSharingGroup(slotSharingGroup);
+
+		sink.connectNewDataSetAsInput(source, DistributionPattern.POINTWISE, resultPartitionType);
+		JobGraph jobGraph = new JobGraph(source, sink);
+		jobGraph.setScheduleMode(scheduleMode);
+
+		ExecutionConfig executionConfig = new ExecutionConfig();
+		executionConfig.setExecutionMode(executionMode);
+		jobGraph.setExecutionConfig(executionConfig);
+
+		return jobGraph;
+	}
+
+	/**
+	 * Test source that emits {@link LongValue} to downstream.
+	 */
+	public static final class LongValueSource extends AbstractInvokable {
+
+		public LongValueSource(Environment environment) {
+			super(environment);
+		}
+
+		@Override
+		public void invoke() throws Exception {
+			ResultPartitionWriter resultPartitionWriter = getEnvironment().getWriter(0);
+			RecordWriterBuilder<LongValue> recordWriterBuilder = new RecordWriterBuilder<>();
+			if (getEnvironment().getExecutionConfig().getExecutionMode() == ExecutionMode.PIPELINED) {
+				// enable output flush for pipeline mode
+				recordWriterBuilder.setTimeout(100);
```

Comment:
Can we use the default timeout?

Reply:
The default value is defined in StreamConfig and can be accessed directly.